### ISSUE TITLE
Analog output is distortion when play A2DP sink (IDFGH-13964)

### ISSUE THREAD

**Issue by eiffelpeter** [espressif/esp-idf#14795]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.4-dev-3602-ga97a7b0962

### Espressif SoC revision.

Chip is ESP32D0WDQ5 (revision 3)

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### Development Kit.

esp32 doit devkit

### Power Supply used.

USB

### What is the expected behavior?

analog audio output is ok

### What is the actual behavior?

analog audio output is distortion

### Steps to reproduce.

1. cd examples/bluetooth/bluedroid/classic_bt/a2dp_sink.
2. idf.py build and flash. 
3. connect esp32 i2s with PCM5102.  
![IMG_8908](https://github.com/user-attachments/assets/bfb3a152-1fc2-4c11-b5c9-d2b423ab6806)

5. iphone connect esp32 via BT. 
6. play 0dB 1kHz on iphone. 
[1kHz 0dB.zip](https://github.com/user-attachments/files/17538338/1kHz.0dB.zip)

7. analog is output, but measure it on scope is distortion ( both distortion on R, L channel ).
![TEK00049](https://github.com/user-attachments/assets/849c2fba-89b7-452b-ae9b-feeeba91578e)

8. I try to **half** data that received from a2dp.  
```
void reduce_data(uint8_t *data, uint32_t len)
{
	uint32_t i;
	uint16_t tmp;

	for (i=0; i<len; i++) {
		tmp = (data[i+1] << 8) | data[i];
		tmp = tmp >> 1;
		data[i] = tmp & 0xFF;
		data[i+1] = tmp >> 8;

		tmp = (data[i+3] << 8) | data[i+2];
		tmp = tmp >> 1;
		data[i+2] = tmp & 0xFF;
		data[i+3] = tmp >> 8;

		i = i+3;
	}
}

void bt_app_a2d_data_cb(const uint8_t *data, uint32_t len)
{
	reduce_data(data, len);
        write_ringbuf(data, len);
}
```

9. The 1kHz is output correctlly.  
![TEK00047](https://github.com/user-attachments/assets/ffbd948f-faf6-4872-ba2e-4af8de9ed241)

also meet Ti 2.1 Vrms
![未命名](https://github.com/user-attachments/assets/8788b668-e9bd-4212-b194-1c37e41b3f18)

10. I do not know is it correct solution ?

**Comment by Grahamauld** [comment]
I found your bug report while investigating the same issue with the PCM5102. I had configured my PCM5102 (FMT pin) for I2S (low) - got exactly the same issue. 

Looking at the "Standard Mode" references here https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/i2s.html I figured I had a bit offset - the default config in the code demo is MSB format (no offset).
The datasheet for the PCM5102 section 9.3.2.2 says that I2S format has an offset between WS/LRCLK and the first bit and that you need LSB format for no offset.

I initially (to save soldering) tried to replace in main/bt_app_av.c
.slot_cfg = I2S_STD_MSB_SLOT_DEFAULT_CONFIG(I2S_DATA_BIT_WIDTH_16BIT, I2S_SLOT_MODE_STEREO),
with
.slot_cfg = I2S_STD_PHILIPS_SLOT_DEFAULT_CONFIG(I2S_DATA_BIT_WIDTH_16BIT, I2S_SLOT_MODE_STEREO),
as mentioned in https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/peripherals/i2s.html

I had expected that to change to I2S(bit shifted mode) and hence match the hardware -  it appeared to make no difference when I changed that and reprogrammed.

Eventually I swapped the FMT pin to high on the DAC so it expected left justified (MSB format) data and it worked perfectly as expected.

TLDR; swap your FMT setting in hardware - dunno why I couldn't fix it in software though so maybe still a bug
